ADC/Core/Src/init.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * init.c
+ *
+ *  Created on: Mar 10, 2020
+ *      Author: rodrigocelmer
+ */
+#include "stm32l053xx.h"
+
+void ADC_Init (void)
+{
+	/*
+	 * since we're going to initialize and then calibrate the ADC, we have to be sure that:
+	 * 1) ADEN = 0,
+	 * 2) DMAEN = 0,
+	 * 3) ADSTART = 0 and
+	 * 4) ADRDY = 0
+	 */
+	if(ADC1->CR & ADC_CR_ADSTART)			//checking if there's an ongoing conversion (3)
+	{
+		ADC1->CR |= ADC_CR_ADSTP;			//if so, we'll stop it
+		while(ADC1->CR & ADC_CR_ADSTP);
+	}
+	ADC1->CR |= ADC_CR_ADDIS;				//disables the ADC (1)
+	while(ADC1->CR & ADC_CR_ADEN);
+	ADC1->CFGR1 &= ~ADC_CFGR1_DMAEN;		//disables DMA (2)
+	if(ADC1->ISR & ADC_ISR_ADRDY)			//and we make sure ADRDY = 0 (4)
```

Comment:
instead of checking ADRDY status and then clearing it if necessary, we can clear it directly.